### PR TITLE
Pass Editor: Create placeholder project and app based off of MC

### PR DESCRIPTION
**Pull request by gadams3**
## What does this PR do?

This change creates a project for the pass editor based off of the material canvas code base. Since many of the core components for material canvas were moved to atom tools framework, the pass editor is bare bones with all of the features of material canvas except for a library of pass related nodes and the code for adapting or transforming pass related data. Other elements like the viewport and main window may also be consolidated.

The largest part of making this tool operational should be reflecting all of the pass related types to edit and behavior contexts. Except for hardcoded C++ passes, all of the pass related data types are already reflected to the serialize context, configured and loaded from JSON based assets.

Rather than creating node configuration files, it should be possible to read the data from the pass system and source files to create the node definitions from them.

It might also be possible to enumerate data through the serialize context to avoid the other reflection.

TBD

Signed-off-by: gadams3 <guthadam@amazon.com>

## How was this PR tested?

Pass editor builds and runs, provides all of the menus, windows, types, and functionality of material canvas except for the nodes and other material related data types.

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Editor/Scripts/bootstrap.py [hunk 1]
```diff
@@ -394,6 +402,9 @@ def bootstrap_AssetBuilder():
         elif O3DE_EDITOR.stem.lower() == "materialcanvas":
             _settings = bootstrap_MaterialEditor()
 
+        elif O3DE_EDITOR.stem.lower() == "passcanvas":
+            _settings = bootstrap_MaterialEditor()
+
```
Comment (rgba16f) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Editor/Scripts/bootstrap.py:407]: Copy and past error, wrong needs to be `bootstrap_PassCanvas`

**Review by rgba16f — APPROVED** [review]

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Editor/Scripts/bootstrap.py (hunk 1 above)
Comment (gadams3) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Editor/Scripts/bootstrap.py:407]: I don't think it's an error. I'll have to confirm with Jonny. There are multiple instances of bootstrap_MaterialEditor(), which I think doesn't do anything.

Comment (rgba16f) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Editor/Scripts/bootstrap.py:407]: This PR adds `bootstrap_PassCanvas`

File: Gems/Atom/Tools/PassCanvas/Code/Source/Window/PassCanvasMainWindow.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ */
+
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <AtomToolsFramework/Document/AtomToolsDocumentInspector.h>
+#include <AtomToolsFramework/Document/AtomToolsDocumentMainWindow.h>
+#include <AtomToolsFramework/EntityPreviewViewport/EntityPreviewViewportSettingsInspector.h>
+#include <AtomToolsFramework/EntityPreviewViewport/EntityPreviewViewportToolBar.h>
```
Comment (jeremyong-az) [inline Gems/Atom/Tools/PassCanvas/Code/Source/Window/PassCanvasMainWindow.h:15]: I noticed this in a few headers now but I believe many of these headers can be replaced by forward declarations instead

File: Gems/Atom/Tools/PassCanvas/Code/Source/Window/PassCanvasViewportContent.cpp [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+namespace PassCanvas
+{
+    PassCanvasViewportContent::PassCanvasViewportContent(
+        const AZ::Crc32& toolId,
+        AtomToolsFramework::RenderViewportWidget* widget,
+        AZStd::shared_ptr<AzFramework::EntityContext> entityContext)
+        : AtomToolsFramework::EntityPreviewViewportContent(toolId, widget, entityContext)
+    {
+        // Configure tone mapper
+        m_postFxEntity = CreateEntity(
```
Comment (jeremyong-az) [inline Gems/Atom/Tools/PassCanvas/Code/Source/Window/PassCanvasViewportContent.cpp:43]: I imagine some of this code will be replicated/needed from the material tools. Does it make sense to have a shared set of library routines like this one to set up the lighting environment?

File: Gems/Atom/Tools/PassCanvas/Code/Source/main.cpp [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+#include <PassCanvasApplication.h>
+
+int main(int argc, char** argv)
+{
+    const AZ::Debug::Trace tracer;
+    AzQtComponents::AzQtApplication::InitializeDpiScaling();
+
+    PassCanvas::PassCanvasApplication app(&argc, &argv);
```
Comment (jeremyong-az) [inline Gems/Atom/Tools/PassCanvas/Code/Source/main.cpp:16]: Question, why do we not just pass `argc` and `argv` directly? Convention?

**Review by jeremyong-az — APPROVED** [review]

**Review by gadams3 — COMMENTED** [review]
Trying to resolve pending comments

**Pull request merged**